lib/tts_isolate.dart

```diff
@@ -43,10 +45,30 @@ Future<bool> onIosBackground(ServiceInstance service) async {
 void onStart(ServiceInstance service) async {
   DartPluginRegistrant.ensureInitialized();
   WidgetsFlutterBinding.ensureInitialized();
-
-  // Initialize Firebase
   await Firebase.initializeApp();
 
+  final prefs = await StreamingSharedPreferences.instance;
+  prefs
+      .getString('tts_channel', defaultValue: '{}')
+      .switchMap((channel) {
+        if (channel.isNotEmpty && channel != "{}") {
+          return FirebaseFirestore.instance
+              .collection('channels')
+              .where('channelId', isEqualTo: channel)
+              .snapshots();
+        } else {
+          return Stream.empty();
+        }
+      })
+      .switchMap((snapshot) => Stream.fromIterable(snapshot.docChanges))
+      .listen((change) {
+        if (change.type == DocumentChangeType.added) {
+          var message = change.doc.data();
+          vocalizeMessage(message);
+        }
+      });
```

Comment:
```suggestion
      .listen((change) {
        for (final change in snapshot.docChanges) {
          if (change.type == DocumentChangeType.added) {
            vocalizeMessage(change.doc.data());
          }
        }
      });
```

There's no need to convert to a stream here.